### PR TITLE
Supporting parsing of probe_messages

### PR DESCRIPTION
**Pull request by hk-spark**
### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [x] Associated README.md updated.
- [x] Has appropriate unit tests.

Today probe messages are not parsed and hence never sent out to influx/promethus. Adding explicit change for fixing this.

### CODE REVIEW

**Comment by hk-spark** [comment]
```
probe_message,device_key=time_series,device=time_series,host=b720273492cf property="[name:\"service_name\" value:\"interface\"  name:\"system_id\" value:\"5254008BA2FC\"  name:\"key\" value:\"Ethernet1\" ]",value="&{0}",probe_id="3ba6ea6b-d472-42b1-aa4c-411e9006b353",stage_name="leaf_fab_if_flap_anomaly" 1524768322713599115

probe_message,device_key=time_series,device=time_series,host=b720273492cf probe_id="3ba6ea6b-d472-42b1-aa4c-411e9006b353",stage_name="leaf_fab_if_flap_anomaly",property="[name:\"service_name\" value:\"interface\"  name:\"system_id\" value:\"525400FCF974\"  name:\"key\" value:\"Ethernet1/3\" ]",value="&{0}" 1524768322713826925

probe_message,device_key=time_series,device=time_series,host=b720273492cf property="[name:\"service_name\" value:\"interface\"  name:\"system_id\" value:\"525400FCF974\"  name:\"key\" value:\"Ethernet1/1\" ]",value="&{0}",probe_id="3ba6ea6b-d472-42b1-aa4c-411e9006b353",stage_name="leaf_fab_if_flap_anomaly" 1524768322713887426
```

File: plugins/inputs/aos/aos.go [hunk 1]
```diff
@@ -335,6 +336,37 @@ func (ssl *streamAos) msgReader(r io.Reader) {
 				}
 			}
 
+
+			if newProbeMessage != nil {
+
+				// Extract device name from Interface name
```
Comment (Ragsboss) [inline plugins/inputs/aos/aos.go]: Remove commented code.

File: plugins/inputs/aos/aos.go [hunk 2]
```diff
@@ -335,6 +336,37 @@ func (ssl *streamAos) msgReader(r io.Reader) {
...
+				// Prepare value. type and property
+				myValue := reflect.ValueOf(newProbeMessage).Elem()
+				myType := myValue.Type()
+				propType := proto.GetProperties(myType)
+
+				serie := "probe_message"
+				fields := make(map[string]interface{})
+				tags := ssl.GetTags( originName )
+
+				// tags["interface"] = devInt
```
Comment (Ragsboss) [inline plugins/inputs/aos/aos.go]: Remove line

File: plugins/inputs/aos/aos.go [hunk 3]
```diff
@@ -335,6 +336,37 @@ func (ssl *streamAos) msgReader(r io.Reader) {
...
+
+				for i := 0; i < myValue.NumField(); i++ {
+
+						myField := myValue.Field(i)
+						field_name := propType.Prop[i].OrigName
+
+						if strings.Contains(field_name, "XXX_") {	continue	}
+
+						fields[propType.Prop[i].OrigName] = reflect.Indirect(myField).Interface()
```
Comment (Ragsboss) [inline plugins/inputs/aos/aos.go]: Use `field_name` instead?

File: plugins/inputs/aos/aos.go [hunk 4]
```diff
@@ -335,6 +336,37 @@ func (ssl *streamAos) msgReader(r io.Reader) {
 				}
 			}
 
+
+			if newProbeMessage != nil {
```
Comment (Ragsboss) [inline plugins/inputs/aos/aos.go:365]: @huttmf I recall we were going to add test code to this? This is a classic change that would benefit from a unit-test..

**Review by Ragsboss — CHANGES_REQUESTED** [review]

**Comment by hk-spark** [comment]
@huttmf @Ragsboss Can we merge this ?

File: plugins/inputs/aos/aos.go (hunk 4 above)
Comment (hk-spark) [reply plugins/inputs/aos/aos.go:365]: @Ragsboss @huttmf Added unit tests now. 

I am still working on polishing it further by supporting msg reader .. but it requires structuring of the messages in groups and a few things like that. Will take time.

File: plugins/inputs/aos/aos.go [hunk 5]
```diff
@@ -155,7 +155,32 @@ func (ssl *streamAos) GetTags(deviceKey string) map[string]string {
...
+	myValue := reflect.ValueOf(newProbeMessage).Elem()
+	myType := myValue.Type()
+	propType := proto.GetProperties(myType)
+
+	serie := "probe_message"
+	fields := make(map[string]interface{})
+	tags := ssl.GetTags( originName )
+
+	for i := 0; i < myValue.NumField(); i++ {
+			myField := myValue.Field(i)
```
Comment (Ragsboss) [inline plugins/inputs/aos/aos.go]: Fix the indent? Just 4 spaces relative to for loop

File: plugins/inputs/aos/aos.go [hunk 6]
```diff
@@ -155,7 +155,32 @@ func (ssl *streamAos) GetTags(deviceKey string) map[string]string {
...
+			field_name := propType.Prop[i].OrigName
+
+			if strings.Contains(field_name, "XXX_") {	continue	}
```
Comment (Ragsboss) [inline plugins/inputs/aos/aos.go]: Remove too many spaces around continue

File: plugins/inputs/aos/aos.go [hunk 7]
```diff
@@ -179,8 +204,8 @@ func (ssl *streamAos) msgReader(r io.Reader) {
 			return
 		}
 
-		msgReader := io.LimitReader(r, int64(msgSize))
-		msgBuf, err := ioutil.ReadAll(msgReader)
+		MsgReader := io.LimitReader(r, int64(msgSize))
```
Comment (Ragsboss) [inline plugins/inputs/aos/aos.go]: We have a var name that is same as method name? can we use different strings?

File: plugins/inputs/aos/aos.go [hunk 8]
```diff
@@ -335,6 +361,31 @@ func (ssl *streamAos) msgReader(r io.Reader) {
 				}
 			}
 
+
+			if newProbeMessage != nil {
+				ssl.ExtractProbeData( newProbeMessage, originName)
+				// Prepare value. type and property
```
Comment (Ragsboss) [inline plugins/inputs/aos/aos.go]: Remove commented code.

**Review by Ragsboss — APPROVED** [review]
Few minor comments..

**Comment by hk-spark** [comment]
@huttmf if you are good with this, I'll go ahead and merge.

**Review by huttmf — APPROVED** [review]
Ok - LGTM

**Pull request merged**